packages/snap/src/helpers/credential-helpers.ts

```diff
@@ -1,48 +1,42 @@
 import { Claim } from '@iden3/js-iden3-core';
 import { ProofType, StorageKeys } from '../enums';
-import { getItemFromStore, setItemInStore } from '../rpc';
 import {
   CredentialStatus,
   ProofQuery,
   RevocationStatus,
   W3CCredential,
 } from '../types';
+import { getItemFromStore, setItemInStore } from '../rpc';
 import { StandardJSONCredentialsQueryFilter } from './json-query-helpers';
 import { getCoreClaimFromProof } from './proof-helpers';
+import {
+  getDecryptedCredentials,
+  saveEncryptedCredentials,
+} from './ceramic-helpers';
+import { uniqBy } from './common-helpers';
 
-export const saveCredential = async (
-  key: string,
-  value: W3CCredential,
+export const saveCredentials = async (
+  credentials: W3CCredential[],
   keyName = 'id' as keyof W3CCredential,
 ): Promise<void> => {
-  const data = (await getItemFromStore(
-    StorageKeys.credentials,
-  )) as W3CCredential[];
-  const items = data || [];
-
-  const itemIndex = items.findIndex((i) => i[keyName] === key);
-  if (itemIndex === -1) {
-    items.push(value);
-  } else {
-    items[itemIndex] = value;
-  }
-  await setItemInStore(StorageKeys.credentials, items);
+  const data = await getDecryptedCredentials();
+  const items = uniqBy([...credentials, ...data], keyName);
+  await saveEncryptedCredentials(items);
 };
 
-export const saveCredentials = async (
-  value: W3CCredential[],
-  keyName = 'id' as keyof W3CCredential,
-): Promise<void> => {
-  for (const credential of value) {
-    await saveCredential(credential.id, credential, keyName);
+export const moveStoreVCtoCeramic = async () => {
+  const credentials = (await getItemFromStore(StorageKeys.credentials)) || [];
+  if (credentials.length) {
```

Comment:
This might cause problems if the user had the snap on multiple devices. Let's do it this way:

```ts
if (credentials.length === 0) {
  return;
}

let ceramicCredentials = await fetchCredentials();

await saveCredentials(merge(credentials, ceramicCredentials));
await setItemInStore(StorageKeys.credentials, []);
```

Reply:
we merge it in the `save Credentials` method